dev/com.ibm.ws.kernel.service/src/com/ibm/ws/common/crypto/CryptoUtils.java

```diff
@@ -0,0 +1,382 @@
+        if (isRunningBetaMode()) {
+            Tr.warning(tc, "CRYPTO_INSECURE_PROVIDER", provider, insecureAlgorithm);
+        }
+    }
+
+    public static boolean isIBMJCEAvailable() {
+        if (ibmJCEProviderChecked) {
+            return ibmJCEAvailable;
+        } else {
+            ibmJCEAvailable = JavaInfo.isSystemClassAvailable(IBMJCE_PROVIDER);
+            ibmJCEProviderChecked = true;
+            return ibmJCEAvailable;
+        }
+    }
+
+    public static boolean isIBMJCEPlusFIPSAvailable() {
+        if (ibmJCEPlusFIPSProviderChecked) {
+            return ibmJCEPlusFIPSAvailable;
+        } else {
+            ibmJCEPlusFIPSAvailable = JavaInfo.isSystemClassAvailable(IBMJCE_PLUS_FIPS_PROVIDER);
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "ibmJCEPlusFIPSProvider: " + IBMJCE_PLUS_FIPS_PROVIDER);
+            }
+
+            ibmJCEPlusFIPSProviderChecked = true;
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "ibmJCEPlusFIPSAvailable: " + ibmJCEPlusFIPSAvailable);
+            }
+
+            if (isRunningBetaMode() && ibmJCEPlusFIPSAvailable) {
+                ibmJCEPlusFIPSAvailable = true;
+            } else {
+                if (fipsEnabled && !isSemeruFips()) {
+                    Tr.error(tc, "FIPS is enabled but the " + IBMJCE_PLUS_FIPS_PROVIDER + " provider is not available.");
+                }
+                ibmJCEPlusFIPSAvailable = false;
+            }
+            return ibmJCEPlusFIPSAvailable;
+        }
+    }
+
+    public static boolean isOpenJCEPlusAvailable() {
+        if (openJCEPlusProviderChecked) {
+            return openJCEPlusAvailable;
+        } else {
+            openJCEPlusAvailable = JavaInfo.isSystemClassAvailable(OPENJCE_PLUS_PROVIDER);
+            openJCEPlusProviderChecked = true;
+            return openJCEPlusAvailable;
+        }
+    }
+
+    public static boolean isOpenJCEPlusFIPSAvailable() {
+        if (openJCEPlusFIPSProviderChecked) {
+            return openJCEPlusFIPSAvailable;
+        } else {
+            openJCEPlusFIPSAvailable = JavaInfo.isSystemClassAvailable(OPENJCE_PLUS_FIPS_PROVIDER);
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "openJCEPlusFIPSProvider: " + OPENJCE_PLUS_FIPS_PROVIDER);
+            }
+
+            openJCEPlusFIPSProviderChecked = true;
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "openJCEPlusFIPSAvailable: " + openJCEPlusFIPSAvailable);
+            }
+
+            if (isRunningBetaMode() && openJCEPlusFIPSAvailable) {
+                openJCEPlusFIPSAvailable = true;
+            } else {
+                if (fipsEnabled && isSemeruFips()) {
+                    Tr.error(tc, "Semeru FIPS is enabled but the " + OPENJCE_PLUS_FIPS_PROVIDER + " provider is not available.");
```

Comment:
Similar statement about Tr.error requiring a translated message.

Reply:
This is temporary for now.